neo/Ledger/MemoryPool.cs

```diff
@@ -401,7 +405,7 @@ internal void UpdatePoolForBlockPersisted(Block block, Snapshot snapshot)
 
             // If we know about headers of future blocks, no point in verifying transactions from the unverified tx pool
             // until we get caught up.
-            if (block.Index < Blockchain.Singleton.HeaderHeight)
+            if (block.Index > 0 && block.Index < Blockchain.Singleton.HeaderHeight)
```

Comment:
When persisting the GenesisBlock, block.Index will be 0, We can't yet access Blockchain.Singleton at that point since the constructor of Blockchain will still be running. This fixes that issue.